### PR TITLE
:recycle: ref: SLOs for Slack postMessage

### PR DESCRIPTION
**Pull request by iamrajjoshi**
Adding SLOs for when we send a slack message for issue and metric alerts

Also added missing SLO for getting parent notification

### CODE REVIEW

File: src/sentry/integrations/slack/actions/notification.py [hunk 1]
```diff
@@ -181,57 +186,68 @@ def get_thread_ts(lifecycle: EventLifecycle) -> str | None:
...
-                }
-
-                log_params = {
-                    "error": str(e),
-                    "project_id": event.project_id,
-                    "event_id": event.event_id,
-                    "channel_name": self.get_option("channel"),
-                }
-                # temporarily log the payload so we can debug message failures
-                log_params["payload"] = json_blocks
```
Comment (iamrajjoshi) [inline src/sentry/integrations/slack/actions/notification.py:215]: log from 6 months ago

File: src/sentry/integrations/slack/actions/notification.py [hunk 2]
```diff
@@ -181,57 +186,68 @@ def get_thread_ts(lifecycle: EventLifecycle) -> str | None:
...
+                        tags={
+                            "action": "send_notification",
+                            "ok": e.response.get("ok", False),
+                            "status": e.response.status_code,
+                        },
+                    )
+
+                    lifecycle.add_extras(log_params)
+                    # If the error is a channel not found or archived, we can halt the flow
+                    if unpack_slack_api_error(e) in (CHANNEL_NOT_FOUND, CHANNEL_ARCHIVED):
```
Comment (iamrajjoshi) [inline src/sentry/integrations/slack/actions/notification.py:242]: starting with this, can update after we get some SLOs

File: src/sentry/integrations/slack/actions/notification.py [hunk 3]
```diff
@@ -181,57 +186,68 @@ def get_thread_ts(lifecycle: EventLifecycle) -> str | None:
...
+                        lifecycle.record_halt(e)
```
Comment (GabeVillalobos) [inline src/sentry/integrations/slack/actions/notification.py:243]: I've been finding halts with stacktraces less helpful when searching through logs. Do you think there's value in including it here?

**Review by GabeVillalobos — APPROVED** [review]

File: src/sentry/integrations/slack/actions/notification.py (hunk 3 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/actions/notification.py:243]: yeah i don't think itll be much useful, instead going to put the reason w/o stack trace

**Pull request merged**